notebooks/usage.ipynb

```diff
@@ -0,0 +1,362 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Status (03/22/2024)\n",
+    "- no existing library fulfills all requirements \n",
+    "- core problem: model firmly baked into code everywhere in all of them\n",
+    "- we had to start from a very low level:\n",
+    "  - only able to analyze files\n",
+    "  - core necessary parts:\n",
+    "    - abstractions for recording, preprocessor, model \n",
+    "    - model, preprocessing disentangled from rest => allows runtime replacement of model\n",
+    "  \n",
+    "- currently there: \n",
+    "  - [x] model, preprocessor abstractions \n",
+    "  - [x] disentanglement of data and analysis => allows exchange of analysis part \n",
+    "  - [x] integration of core birdnet functionality back \n",
+    "  - runs in python from prerecorded files \n",
+    "  \n",
+    "- not yet there: \n",
+    "  - user interface \n",
+    "  - integrated system that would 'run alone'  \n",
+    "    - `birdnetlib` provides at the very least starting points \n",
+    "  - documentation\n",
+    "  - no support for pytorch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Code example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "clone repo and install first from repo directory `python3 -m pip install .`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys \n",
+    "sys.path.append(\"..\")\n",
+    "import iSparrow.sparrow_model_base as spm\n",
+    "import iSparrow.sparrow_recording as spr\n",
+    "import iSparrow.preprocessor_base as spb\n",
+    "import iSparrow.utils as utils\n",
+    "\n",
+    "import tests.make_mock_install as sp"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "import pandas as pd"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# make a mock install of sparrow. will be invisible in the future\n",
+    "sp.install()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Demo\n",
+    "\n",
+    "- load model, preprocessor you want \n",
+    "- add recording to be analyzed and assign it the model, preprocessor to use \n",
+    "- analyze "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# variables for analysis\n",
+    "sigmoid_sensitivity = 1.0\n",
+    "num_threads = 12\n",
+    "min_conf = 0.25\n",
+    "\n",
+    "# variables for recording\n",
+    "recording_path = sp.EXAMPLES / \"soundscape.wav\"\n",
```

Comment:
why do we set it here and not use it later?

Reply:
ok, no idea why that happened. I must have just forgotten about that variable.  Fixed.